src/App.tsx

```diff
@@ -1,14 +1,40 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({ query: value });
+  };
 
   render() {
+    const { query } = this.state;
+
+    const visibleMovies = moviesFromServer.filter(movie => {
+      const lowerTitle = movie.title.toLowerCase();
+      const lowerDescription = movie.description.toLowerCase();
+      const lowerQuery = query.toLowerCase();
+
+      if (lowerTitle.includes(lowerQuery)
+        || lowerDescription.includes(lowerQuery)) {
+        return movie;
+      }
+
+      return null;
```

Comment:
filter's callback should return true or false. 
why not just use the result of the includes method? 

```suggestion
      return lowerTitle.includes(lowerQuery) || lowerDescription.includes(lowerQuery);
```